DataDefinitions/StarSystem.cs

```diff
@@ -130,7 +131,27 @@ public class StarSystem
 
         // Not intended to be user facing - materials available within the system
         [JsonIgnore]
-        public List<string> materialEdNames => bodies.SelectMany(b => b.materials, (b, m) => m.definition.edname).Distinct().ToList();
+        public List<string> materialEdNames
+        {
+            get
+            {
+                try
+                {
+                    List<Body> bodiesList = null;
+                    lock (bodyLock)
+                    {
+                        bodiesList = bodies;
+                    }
+                    return bodiesList.SelectMany(b => b.materials, (b, m) => m.definition.edname).Distinct().ToList();
+                }
+                catch (InvalidOperationException ex)
+                {
+                    Logging.Error("Error aggregating system materials list.", ex);
+                    return new List<string>();
+                }
+            }
+        }
+        private readonly object bodyLock = new object();
```

Comment:
It doesn't look like changes to `bodies` are guarded by `bodyLock`. 

What is the exact concurrency issue we are looking to solve here?

Reply:
Once we've obtained a copy of the `bodies` collection, we can release exclusive access to that collection and build our list of material ednames from the copy without a risk of collision.

Reply:
It is very important to understand the the lock introduced here does not in any way enforce "exclusive access" to the collection. It merely means that the few lines within that lock section cannot be entered concurrently on differerent threads. It is still entirely possible for code elsewhere, running in another thread, to modify the collection while the copy constructor is in flight and then we will get the same exception again.

Reply:
Agreed. Lock removed.